src/Illuminate/Support/Str.php

```diff
@@ -253,6 +253,42 @@ public static function endsWith($haystack, $needles)
         return false;
     }
 
+    /**
+     * Extracts an excerpt from text that matches the first instance of phrase.
+     *
+     * @param  string  $text
+     * @param  string  $phrase
+     * @param  array  $options
+     * @return string|null
+     */
+    public static function excerpt($text, $phrase = '', $options = [])
+    {
+        $radius = $options['radius'] ?? 100;
+        $omission = $options['omission'] ?? '...';
+
+        preg_match_all("/^(\W*|.*)(".preg_quote((string) $phrase).")(.*\W*)$/i", (string) $text, $matches);
```

Comment:
```suggestion
        preg_match("/^(.*?)(".preg_quote((string) $phrase).")(.*)$/i", (string) $text, $matches);
```

Regex can be simplified to get rid of `\W` and I added a lazy token to the first match, to remove the need for `preg_match_all`.

This also means that the following statements that uses `$matches`  don't be wrapped in a `collect#first`.